src/static/js/chat.js

```diff
@@ -199,14 +190,11 @@ var chat = (function()
 
           if(!chatOpen && ctx.duration > 0) {
             $.gritter.add({
-              // (string | mandatory) the heading of the notification
-              title: ctx.authorName,
-              // (string | mandatory) the text inside the notification
-              text: ctx.text,
-              // (bool | optional) if you want it to fade out on its own or just sit there
+              text: '<span class="author-name">' + ctx.authorName + '</span>' + ctx.text,
```

Comment:
I do not know this library, I may be wrong.

What use is this library going to do of `text`? Is it possible that `ctx.authorName` and `ctx.text` end up being appended to the Dom without any escape? If so, this would end up being a nice XSS vulnerability.

To test it, one could force:

```js
ctx.authorName = '<script>alert(1);</script>';
ctx.text = '<script>alert(1);</script>';
```

somewhere in that function, and find a way of triggering it.
It's a chat module, maybe send some message?

Reply:
Yep, there is an XSS.

![xss-pr-3744](https://user-images.githubusercontent.com/4067621/79673795-295df380-81dd-11ea-85d6-0a7b75bc81c5.gif)

@seballot: the universal takeaway is always be wary of string concatenation :smile:

Reply:
I think in this case someone has already taken care of it. Those two fields of the `ctx` variable are already escaped:
https://github.com/ether/etherpad-lite/blob/14be1defb6757c9c3fa900a29f8b9f179a5172ad/src/static/js/chat.js#L142-L144

Probably someone had already discovered a vulnerability there.

Please note that the fix that we have in palce is **not** good: it merely escapes two attributes and passes them around. The escaping should not be done at that level, but on the view, just before generating it.

@seballot: not a problem here. In sane project taking care of that escaping should be responsibility of who writes the views, because that is the correct layer to address the problem. In this case "we" (the developers) are saying to you that - yes - you can trust our strings to always be escaped. It does not work like this... :smile: